limacharlie/rule.go

```diff
@@ -0,0 +1,139 @@
+package limacharlie
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type DRRuleOptions struct {
+	// Replace rule if it already exists with this name.
+	IsReplace bool
+	// Rule namespace, defaults to "general".
+	Namespace string
+	// Rule is enabled.
+	IsEnabled bool
+	// Number of seconds before rule auto-deletes.
+	TTL int64
+}
+
+type drAddRuleRequest struct {
+	Name      string `json:"name"`
+	IsReplace bool   `json:"is_replace,string"`
+	Detection string `json:"detection"`
+	Response  string `json:"response"`
+	IsEnabled bool   `json:"is_enabled,string"`
+	ExpireOn  int64  `json:"expire_on,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type CoreDRRule struct {
+	Name      string                   `json:"name"`
+	Namespace string                   `json:"namespace,omitempty"`
+	Detect    map[string]interface{}   `json:"detect"`
+	Response  []map[string]interface{} `json:"respond"`
+}
+
+// DRRuleAdd add a D&R Rule to an LC organization
+func (org Organization) DRRuleAdd(name string, detection interface{}, response interface{}, opt ...DRRuleOptions) error {
```

Comment:
Do we really want to support passing multiple `DRRuleOptions`? It feels like a pattern to avoid having to pass an empty `DRRuleOptions`. I would rather prefer seeing something like `DRRuleAdd` with a mandatory parameter for options and one called something like `DRRuleAddNoOptions`. What do you think?

Reply:
It's a pretty common pattern in a lot of libs that require clients with options, not to provide multiple values but just using the variadic as an optional single var.

I generally prefer that to using empty dummy options since those non-useful code. There are however alternative patterns we could use like variadic optional configuration functions: https://petomalina.medium.com/dealing-with-optional-parameters-in-go-9780f9bfbd1d

In the case of options adding DR rules, variadic functions feels heavy handed since we're not dealing with complex highly-customizable behavior, just a couple of flags.

For the `DRRules()` call however, it might eventually support more customizable filters, so maybe we could do that there. I'll add it to the PR to see what it would look like.

Reply:
PTAL